### ISSUE TITLE
Every page you land on should have a specific unique url

### ISSUE THREAD

**Issue by twinkietoes-on** [LibriVox/librivox-catalog#52]
Jo Smallheer commented on Saturday 28th September 2013 at 6:12pm
 forum thread discussion links: 
 https://forum.librivox.org/viewtopic.php?f=50&t=47498
 https://forum.librivox.org/viewtopic.php?p=906760#p906760
someone will need to sort through what is done and what still needs to be done in these threads
 
 Hugh M commented on Wednesday 2nd October 2013 at 10:25pm
 jo this seems to be underway already is that correct? do we need this ticket?
 
 Hugh M commented on Wednesday 2nd October 2013 at 10:28pm
 just tested:
 
 browse: language
 browse: genre
  both need to be urlified.  assign to jeff?
 
 Jo Smallheer commented on Thursday 3rd October 2013 at 2:48am 
 The browse pages all still need this. All browse pages still are http://librivox.org/search
  Each page should have a unique url that indicates where and what it is.
 
 Jeff - It's up to you what the actual links should be. I'm fully aware that the following sample links are not what you will actually use for the urls. They are just meant as examples.
 
 Example 1:
 Click on Language to browse by language
 
 URL should change from http://librivox.org/search to something like http://librivox.org/browse/language&page=#
 
 On above page, Click on French and URL should change to something like http://librivox.org/browse/language/french&page=#
 
 Example 2:
 Click on Genre/Subject to browse by Genre
 
 URL should change from http://librivox.org/search to something like http://librivox.org/browse/genre&page=#
 
 On above page click on *Non-fiction > History > Middle Ages/Middle History and url should change to something like http://librivox.org/browse/nonfiction/history/middle_ages/middle_history&page=#
 
 All the Browse pages for Author, Title, Language and Genre need this type of url. Language and Genre also need the sub-types. Titles and Authors just need top level and page #.
 
  Unknown User changed Assignee from Jo Smallheer → Jeff Madsen on Thursday 3rd October 2013 at 2:48am.
 
 Jeff Madsen commented on Saturday 26th October 2013 at 7:29am
 
 I can set up for specific language & author like this - it makes zero sense to do this for titles & authors as they are dynamic data, so saying "go to p5 of titles" will never give you the same results after a few items are added each day/week. We do this sort of thing when we can base on permanent keys
 
 Jo Smallheer commented on Sunday 27th October 2013 at 9:27pm
 
 Please go ahead with Language and Genre pages.
 
 As for the rest, I disagree that it makes no sense to assign each page it's own unique url, regardless if the content on that page is always the same or not. It lets the user navigate using the back and forward buttons when browsing. Am I missing something about how else one can back and forth through the various pages viewed in a given session?

  Jeff Madsen commented on Sunday 27th October 2013 at 10:42pm
  Well, I would suggest the pagination buttons; however, thought you meant they would be sharing these links, not that big a deal to add if there is a reason for it. Just don't want to add everything under the sun if somethings don't work as people are thinking
 
 Jeff Madsen commented on Wednesday 30th October 2013 at 5:32am
  Urls generate everywhere; however, the back & forward browser buttons won't work & I don't see any way to make them do so without rewriting the entire catalog pages.
  Basically, this is a web app, not a web page. They need to use the in-app navigation to get around; even if I could figure out a way around this, it would only work in the latest browsers anyway, so that rules out most of the people who want it
